mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/spi/PojoMassIndexingMappingContext.java

```diff
@@ -6,21 +6,16 @@
  */
 package org.hibernate.search.mapper.pojo.massindexing.spi;
 
-import org.hibernate.search.engine.backend.common.spi.EntityReferenceFactory;
 import org.hibernate.search.engine.backend.mapping.spi.BackendMappingContext;
 import org.hibernate.search.engine.reporting.FailureHandler;
 import org.hibernate.search.engine.environment.thread.spi.ThreadPoolProvider;
+import org.hibernate.search.mapper.pojo.common.spi.PojoEntityReferenceFactoryDelegate;
 
 /**
  * Contextual information about a search mapping.
  */
 public interface PojoMassIndexingMappingContext extends BackendMappingContext {
 
-	/**
-	 * @return A {@link EntityReferenceFactory} that relies on the object's class to return entity types.
-	 */
-	EntityReferenceFactory<?> entityReferenceFactory();
```

Comment:
Since this is implemented by mappers, should we convert this to a default method throwing an exception and deprecate it? (assuming someone created their own mapper 😄) Or because we are the callers of this method -- we can just drop it.

Reply:
This method was just moved to the superinterface `BackendMappingContext`, it's not a breaking change.

That being said there are probably plenty of breaking changes in mapper SPIs. I gave up on ensuring backwards compatibility and just documented the change (since we're allowed to break SPIs in minors).